src/components/Header/utils.ts

```diff
@@ -0,0 +1,265 @@
+import { Navigation, Page, Post } from '@/payload-types'
+
+export type NavLink =
+  | {
+      type: 'internal-reference'
+      label: string
+      reference:
+        | { relationTo: 'pages'; value: number | Page }
+        | { relationTo: 'posts'; value: number | Post }
```

Comment:
Yeah, looks like `value` is a reference if it's not a filled-in object, so using the number as the slug won't work in the code above - likely better to use the log/throw bail-out.